File: CS/RequestDataLoadingExample/Form1.cs
using DevExpress.XtraMap;
using System;
using System.Windows.Forms;


namespace RequestDataLoadingExample {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }
        private void Form1_Load(object sender, EventArgs e) {
            ImageLayer layer = new ImageLayer();
            mapControl1.Layers.Add(layer);
            BingMapDataProvider provider = new BingMapDataProvider();
            provider.BingKey = "YOUR_BING_KEY";
            layer.DataProvider = provider;
            layer.RequestDataLoading += OnRequestDataLoading;
            layer.DataLoaded += OnDataLoaded;
        }
        void OnRequestDataLoading(object sender, EventArgs e) {
            Cursor.Current = Cursors.WaitCursor;
            if (!splashScreenManager1.IsSplashFormVisible)
                splashScreenManager1.ShowWaitForm();
        }
        void OnDataLoaded(object sender, DataLoadedEventArgs e) {
            Cursor.Current = Cursors.Default;
            splashScreenManager1.CloseWaitForm();
        }

    }
}

File: VB/RequestDataLoadingExample/Form1.vb
Imports DevExpress.XtraMap
Imports System
Imports System.Windows.Forms


Namespace RequestDataLoadingExample
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub
        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            Dim layer As New ImageLayer()
            mapControl1.Layers.Add(layer)
            Dim provider As New BingMapDataProvider()
            provider.BingKey = "YOUR_BING_KEY"
            layer.DataProvider = provider
            AddHandler layer.RequestDataLoading, AddressOf OnRequestDataLoading
            AddHandler layer.DataLoaded, AddressOf OnDataLoaded
        End Sub
        Private Sub OnRequestDataLoading(ByVal sender As Object, ByVal e As EventArgs)
            Cursor.Current = Cursors.WaitCursor
            If Not splashScreenManager1.IsSplashFormVisible Then
                splashScreenManager1.ShowWaitForm()
            End If
        End Sub
        Private Sub OnDataLoaded(ByVal sender As Object, ByVal e As DataLoadedEventArgs)
            Cursor.Current = Cursors.Default
            splashScreenManager1.CloseWaitForm()
        End Sub

    End Class
End Namespace
